### PR TITLE
Use ruby for editing yaml config file

### PR DESCRIPTION
**Pull request by echlebek**
Making a PR before the tests pass, because I've been banned from the github releases page on my local machine :headstone:

### CODE REVIEW

**Comment by echlebek** [comment]
This has been quite challenging to work with, as I'm getting 403s locally after many cycles of debug/develop/test. However, at this point, most of the tests are passing and probably only one or two small fixes are needed.

File: install-script/install.sh [hunk 1]
```diff
@@ -1367,17 +1395,7 @@ function write_api_url() {
...
-
-    # ToDo: ensure we override only sumologic `api_base_url`
-    if grep "api_base_url" "${file}" > /dev/null; then
-        sed -i.bak -e "s/api_base_url:.*$/api_base_url: $(escape_sed "${api_url}")/" "${file}"
-    else
-        # write api_url on the top of sumologic: extension
-        sed -i.bak -e "1,/sumologic:/ s/sumologic:/sumologic:\\
-\\${ext_indentation}api_base_url: $(escape_sed "${api_url}")/" "${file}"
-    fi
+    write_yaml_config_key "${file}" "extensions sumologic api_base_url" "${api_url}"
```
Comment (swiatekm-sumo) [inline install-script/install.sh:1398]: It'd be more idiomatic to separate the keys with `.` instead of ` `.

Comment (echlebek) [reply install-script/install.sh:1398]: I wouldn't say anything that's happening in this PR is idiomatic, but I'm happy to make this change :+1:

**Review by swiatekm-sumo — COMMENTED** [review]
I'm ok with this approach in general, assuming it does actually work on all of our supported platforms.

**Comment by echlebek** [comment]
I may give up on this. I realized that ruby isn't installed by default on Ubuntu. So, this approach will break the script for a substantial number of users. I've filed an internal issue with my thoughts on the matter.

**Pull request closed without merging**